core/workspace_comment.js

```diff
@@ -0,0 +1,370 @@
+  this.movable_ = movable;
+};
+
+/**
+ * Returns this comment's text.
+ * @return {string} Comment text.
+ * @package
+ */
+Blockly.WorkspaceComment.prototype.getContent = function() {
+  return this.content_;
+};
+
+/**
+ * Set this comment's content.
+ * @param {string} content Comment content.
+ * @package
+ */
+Blockly.WorkspaceComment.prototype.setContent = function(content) {
+  if (this.content_ != content) {
+    Blockly.Events.fire(
+        new Blockly.Events.CommentChange(this, this.content_, content));
+    this.content_ = content;
+  }
+};
+
+/**
+ * Encode a comment subtree as XML with XY coordinates.
+ * @param {boolean=} opt_noId True if the encoder should skip the comment id.
+ * @return {!Element} Tree of XML elements.
+ * @package
+ */
+Blockly.WorkspaceComment.prototype.toXmlWithXY = function(opt_noId) {
+  var element = this.toXml(opt_noId);
+  element.setAttribute('x', Math.round(this.xy_.x));
+  element.setAttribute('y', Math.round(this.xy_.y));
+  element.setAttribute('h', this.height_);
+  element.setAttribute('w', this.width_);
+  return element;
+};
+
+/**
+ * Encode a comment subtree as XML, but don't serialize the XY coordinates.
+ * This method avoids some expensive metrics-related calls that are made in
+ * toXmlWithXY().
+ * @param {boolean=} opt_noId True if the encoder should skip the comment id.
+ * @return {!Element} Tree of XML elements.
+ * @package
+ */
+Blockly.WorkspaceComment.prototype.toXml = function(opt_noId) {
+  var commentElement = goog.dom.createDom('comment');
+  if (!opt_noId) {
+    commentElement.setAttribute('id', this.id);
+  }
+  commentElement.textContent = this.getContent();
+  return commentElement;
+};
+
+/**
+ * Fire a create event for the given workspace comment, if comments are enabled.
+ * @param {!Blockly.WorkspaceComment} comment The comment that was just created.
+ * @package
+ */
+Blockly.WorkspaceComment.fireCreateEvent = function(comment) {
+  if (Blockly.Events.isEnabled()) {
+    var existingGroup = Blockly.Events.getGroup();
+    if (!existingGroup) {
+      Blockly.Events.setGroup(true);
+    }
+    try {
+      Blockly.Events.fire(new Blockly.Events.CommentCreate(comment));
+    } finally {
+      if (!existingGroup) {
+        Blockly.Events.setGroup(false);
+      }
+    }
+  }
+};
+
+/**
+ * Decode an XML comment tag and create a comment on the workspace.
+ * @param {!Element} xmlComment XML comment element.
+ * @param {!Blockly.Workspace} workspace The workspace.
+ * @return {!Blockly.WorkspaceComment} The created workspace comment.
+ * @package
+ */
+Blockly.WorkspaceComment.fromXml = function(xmlComment, workspace) {
+  var info = Blockly.WorkspaceComment.parseAttributes(xmlComment);
+
+  var comment = new Blockly.WorkspaceComment(
+      workspace, info.content, info.h, info.w, info.id);
+
+  var commentX = parseInt(xmlComment.getAttribute('x'), 10);
+  var commentY = parseInt(xmlComment.getAttribute('y'), 10);
+  if (!isNaN(commentX) && !isNaN(commentY)) {
```

Comment:
Would it make sense to translate in only one direction if only one is provided? Instead of not using either direction if both aren't provided...